display_list/display_list_ops.h

```diff
@@ -224,6 +223,7 @@ DEFINE_SET_CLEAR_DLATTR_OP(ColorFilter, ColorFilter, filter)
 DEFINE_SET_CLEAR_DLATTR_OP(ImageFilter, ImageFilter, filter)
 DEFINE_SET_CLEAR_DLATTR_OP(MaskFilter, MaskFilter, filter)
 DEFINE_SET_CLEAR_DLATTR_OP(ColorSource, Shader, source)
+DEFINE_SET_CLEAR_DLATTR_OP(PathEffect, PathEffect, effect)
```

Comment:
As written, DlPathEffect is not compatible with bulk comparing (see the DlOp::equals and all of its overrides for an example). If the DlDashPathEffect would store its intervals inline with the object as the linear gradients do, then it would be bulk-compare-compatible.

Reply:
The comments on the `DisplayListCompare` enum in display_list_ops.h are a good reference for the use of bulk compares on the DisplayList buffer...